interfaces/Portalicious/src/app/utils/locale.ts

```diff
@@ -0,0 +1,48 @@
+import { registerLocaleData } from '@angular/common';
+import { loadTranslations } from '@angular/localize';
+import { xliffToJson } from '~/utils/xliff-to-json';
+
+export const localStorageLocaleKey = 'preferredLanguage';
+
+export function getStoredLanguage(): string {
+  return localStorage.getItem(localStorageLocaleKey) ?? 'en';
+}
+
+export async function initLanguage(): Promise<void> {
+  const locale = getStoredLanguage();
+
+  if (locale === 'en') {
+    // Default behavior, no changes required
+    return;
+  }
+
+  // Fetch XLIFF translation file and transform to JSON format (JSON translations can be used directly)
+  const json = await fetch('/assets/messages.' + locale + '.xlf')
+    .then((r) => r.text())
+    .then((t) => xliffToJson(t));
```

Comment:
Would be interesting to see if we can just use JSON files directly and avoid the hassle of this conversion.

Reply:
https://github.com/global-121/121-platform/pull/5543